internal/datastore/mysql/datastore.go

```diff
@@ -0,0 +1,648 @@
+)
+
+type sqlFilter interface {
+	ToSql() (string, []interface{}, error)
+}
+
+func NewMySQLDatastore(url string, options ...Option) (*Datastore, error) {
+	config, err := generateConfig(options)
+	if err != nil {
+		return nil, fmt.Errorf(errUnableToInstantiate, err)
+	}
+	connector, err := mysql.MySQLDriver{}.OpenConnector(url)
+	if err != nil {
+		return nil, fmt.Errorf("NewMySQLDatastore: failed to create connector: %w", err)
+	}
+	var db *sql.DB
+	if config.enablePrometheusStats {
+		connector, err = instrumentConnector(connector)
+		if err != nil {
+			return nil, fmt.Errorf("NewMySQLDatastore: unable to instrument connector: %w", err)
+		}
+		db = sql.OpenDB(connector)
+		// Create a new collector, the name will be used as a label on the metrics
+		collector := sqlstats.NewStatsCollector("spicedb", db)
+		err := prometheus.Register(collector)
+		if err != nil {
+			return nil, fmt.Errorf(errUnableToInstantiate, err)
+		}
+	} else {
+		db = sql.OpenDB(connector)
+	}
+	db.SetConnMaxLifetime(config.connMaxLifetime)
+	db.SetConnMaxIdleTime(config.connMaxIdleTime)
+	db.SetMaxOpenConns(config.maxOpenConns)
+	db.SetMaxIdleConns(config.maxOpenConns)
+
+	driver := migrations.NewMysqlDriverFromDB(db, config.tablePrefix)
+	queryBuilder := NewQueryBuilder(driver)
+
+	createTxn, _, err := sb.Insert(driver.RelationTupleTransaction()).Values().ToSql()
+	if err != nil {
+		return nil, fmt.Errorf("NewMySQLDatastore: %w", err)
+	}
+
+	// used for seeding the initial relation_tuple_transaction. using INSERT IGNORE on a known
+	// ID value makes this idempotent (i.e. safe to execute concurrently).
+	createBaseTxn := fmt.Sprintf("INSERT IGNORE INTO %s (id) VALUES (1)", driver.RelationTupleTransaction())
+
+	gcCtx, cancelGc := context.WithCancel(context.Background())
+	querySplitter := common.TupleQuerySplitter{
+		Executor:         newMySQLExecutor(db),
+		UsersetBatchSize: config.splitAtUsersetCount,
+	}
+
+	store := &Datastore{
+		db:                       db,
+		driver:                   driver,
+		url:                      url,
+		revisionFuzzingTimedelta: config.revisionFuzzingTimedelta,
+		gcWindowInverted:         -1 * config.gcWindow,
+		gcInterval:               config.gcInterval,
+		gcMaxOperationTime:       config.gcMaxOperationTime,
+		gcCtx:                    gcCtx,
+		cancelGc:                 cancelGc,
+		watchBufferLength:        config.watchBufferLength,
+		createTxn:                createTxn,
+		createBaseTxn:            createBaseTxn,
+		QueryBuilder:             queryBuilder,
+		querySplitter:            &querySplitter,
+	}
+
+	// Start a goroutine for garbage collection.
+	if store.gcInterval > 0*time.Minute {
+		store.gcGroup, store.gcCtx = errgroup.WithContext(store.gcCtx)
+		store.gcGroup.Go(store.runGarbageCollector)
+	} else {
+		log.Warn().Msg("garbage collection disabled in mysql driver")
+	}
+
+	return store, nil
+}
+
+// unlike PSQL, this implementation does not create a transaction - not needed
```

Comment:
Drop the "unlike" comment and instead add a comment inside the method to explain that no transaction is needed and why

Reply:
addressed in https://github.com/authzed/spicedb/pull/525/commits/60aeb997f94e7b21d2b2a272be2fbd18fc0e18e2